sdk/cosmos/azure-cosmos/azure/cosmos/aio/database.py

```diff
@@ -0,0 +1,768 @@
+        Note: it does not check or update the existing container settings or offer throughput
+        if they differ from what was passed into the method.
+
+        :param id: ID (name) of container to read or create.
+        :param partition_key: The partition key to use for the container.
+        :param indexing_policy: The indexing policy to apply to the container.
+        :param default_ttl: Default time to live (TTL) for items in the container. If unspecified, items do not expire.
+        :param populate_query_metrics: Enable returning query metrics in response headers.
+        :param offer_throughput: The provisioned throughput for this offer.
+        :param unique_key_policy: The unique key policy to apply to the container.
+        :param conflict_resolution_policy: The conflict resolution policy to apply to the container.
+        :keyword str session_token: Token for use with Session consistency.
+        :keyword dict[str,str] initial_headers: Initial headers to be sent as part of the request.
+        :keyword str etag: An ETag value, or the wildcard character (*). Used to check if the resource
+            has changed, and act according to the condition specified by the `match_condition` parameter.
+        :keyword ~azure.core.MatchConditions match_condition: The match condition to use upon the etag.
+        :keyword Callable response_hook: A callable invoked with the response metadata.
+        :keyword analytical_storage_ttl: Analytical store time to live (TTL) for items in the container.  A value of
+            None leaves analytical storage off and a value of -1 turns analytical storage on with no TTL.  Please
+            note that analytical storage can only be enabled on Synapse Link enabled accounts.
+        :returns: A `ContainerProxy` instance representing the container.
+        :raises ~azure.cosmos.exceptions.CosmosHttpResponseError: The container read or creation failed.
+        :rtype: ~azure.cosmos.ContainerProxy
+        """
+
+        analytical_storage_ttl = kwargs.pop("analytical_storage_ttl", None)
+        try:
+            container_proxy = self.get_container_client(id)
+            await container_proxy.read(
+                populate_query_metrics=populate_query_metrics,
+                **kwargs
+            )
+            return container_proxy
+        except CosmosResourceNotFoundError:
+            return await self.create_container(
+                id=id,
+                partition_key=partition_key,
+                indexing_policy=indexing_policy,
+                default_ttl=default_ttl,
+                populate_query_metrics=populate_query_metrics,
+                offer_throughput=offer_throughput,
+                unique_key_policy=unique_key_policy,
+                conflict_resolution_policy=conflict_resolution_policy,
+                analytical_storage_ttl=analytical_storage_ttl
+            )
+
+    def get_container_client(self, container):
```

Comment:
The method currently also takes in a ContainerProxy instance, but I will change that to be just a string since it doesn't really make sense to take in a ContainerProxy and then return it again